test/integration/puppetlabs/services/jruby/tasks_test.clj

```diff
@@ -0,0 +1,116 @@
+(ns puppetlabs.services.jruby.tasks-test
+  (:require [clojure.test :refer :all]
+            [puppetlabs.kitchensink.core :as ks]
+            [puppetlabs.services.jruby.jruby-puppet-testutils :as jruby-testutils]
+            [puppetlabs.services.jruby.puppet-environments :as puppet-env]
+            [puppetlabs.services.master.master-core :as mc]
+            [me.raynes.fs :as fs]
+            [cheshire.core :as cheshire]
+            [schema.core :as schema]
+            [schema.test :as schema-test]
+            [puppetlabs.puppetserver.testutils :as testutils]
+            [puppetlabs.trapperkeeper.app :as tk-app]
+            [puppetlabs.trapperkeeper.testutils.bootstrap :as tk-bootstrap]))
+
+(use-fixtures :once schema-test/validate-schemas)
+
+(def TaskOptions
+  {:name schema/Str
+   :module-name schema/Str
+   :metadata? schema/Bool
+   :number-of-files schema/Num})
+
+(defn make-task-dir
+  [env-dir mod-name]
+  (fs/file env-dir "modules" mod-name "tasks"))
+
+(schema/defn gen-task
+  "Assumes tasks dir already exists -- generates a set of task files for a
+  single task."
+  [env-dir task :- TaskOptions]
```

Comment:
it's a little weird when using schema to have a param that has a type share a line with an unschemaed param.